sass/site/_layout.scss

```diff
@@ -21,41 +21,66 @@
 
 // Content area without sidebar.
 .content-area {
-	width: 100%; 
+	width: 100%;
 }
 
 // Content area with sidebar.
 .sidebar .content-area {
 	float: left;
-	width: 66%;	
-	
+	margin: 0 (-$sidebar__width) 0 0;
+	width: $size__site;
+
 	@at-root .sidebar-left#{&} {
 		float: right;
-	}		 	
-	
+		margin: 0 0 0 (-$sidebar__width);
+	}
+
 	@media (max-width: 768px) {
 		border: 0;
 		padding-right: 0;
-		width: 100%; 	
+		width: 100%;
+		margin: 0;
+	}
+}
+
+.sidebar .site-main {
+	margin: 0 $sidebar__width 0 0;
+
+	@at-root .sidebar-left#{&} {
+		margin: 0 0 0 (-$sidebar__width);
+	}
+
+	@media (max-width: 768px) {
+		margin: 0;
 	}
 }
+
 // No Sidebar page layout setting.
 .page-layout-no-sidebar .site-content .content-area {
 	float: none;
-	width: 100%; 
+	width: 100%;
+	margin: 0;
+
+	.site-main {
+		margin: 0;
```

Comment:
Any idea why this is necessary? When is the initial margin being added to `.site-main` ?

Reply:
The same question for the above rule. Is `margin: 0;` necessary in both cases? Where is the initial margin being added?